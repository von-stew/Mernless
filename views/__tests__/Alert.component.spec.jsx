/*
 * This file is part of the CookieScript project.
 *
 * (c) Yasser Ameur El Idrissi <getspookydev@gmail.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

import '@testing-library/jest-dom'
import React from 'react';
import ReactDom from 'react-dom';
import { render } from '@testing-library/react';
import Alert from 'views/components/Alert.component';

var props = {
  className: 'orange',
  message: 'Be Warned',
  details: 'Something not ideal might be happening.'
};

var wrapper = null;

beforeEach(() => {
  wrapper = <Alert className={props.className} message={props.message} details={props.details} />
});

test('should render component without crash', () => {
  const container = document.createElement('div');
  ReactDom.render(wrapper, container);
  ReactDom.unmountComponentAtNode(container);
});

test('should render props data', () => {
  const { getByTestId } = render(wrapper);
  expect(getByTestId(/testing-root/i)).toHaveClass("orange border-l-4 p-4");
  expect(getByTestId(/testing-header/i)).toHaveTextContent("Be Warned");
  expect(getByTestId(/testing-details/i)).toHaveTextContent("Something not ideal might be happening.");
});

test('should renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
