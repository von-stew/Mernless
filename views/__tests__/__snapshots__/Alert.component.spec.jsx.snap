// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<Alert
  className="orange"
  details="Something not ideal might be happening."
  message="Be Warned"
/>
`;
